api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -303,15 +303,18 @@ impl KubernetesInfrastructure {
         &self,
         app_name: &String,
     ) -> Result<(), KubernetesInfrastructureError> {
-        match Api::v1Namespace(self.client())
+        match Api::<V1Namespace>::all(self.client()?)
             .create(
                 &PostParams::default(),
-                namespace_payload(app_name).into_bytes(),
+                &serde_json::from_value(namespace_payload(app_name))?,
```

Comment:
IMHO we should move the conversion into `namespace_payload` so that is returns `V1Namespace`. For the moment we could use `expect("with a good comment")` in those payload functions so that we do not have to worry here.

Reply:
This is so much nicer! 🥰
On that note: there's `kube-derive` to generate payload schemas from spec structs: https://docs.rs/kube-derive/0.48.0/kube_derive/derive.CustomResource.html
However, it would kind of a pain to properly specify the schema for Traefik …

For now, I think I'll leave the ingress and middleware payloads as `Value`s.

Reply:
Resolved with db1244 and 9f8d95